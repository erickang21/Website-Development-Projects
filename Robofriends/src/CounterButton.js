import React,{useState} from "react"

export const CounterButton = () => {
    const [cnt,setCnt] = useState(0)
    return <button id="counter" onClick={() => setCnt(cnt+1)}>{cnt}</button>
}