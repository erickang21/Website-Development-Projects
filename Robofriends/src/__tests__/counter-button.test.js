import { shallow } from "enzyme";
import React from "react"
import { CounterButton } from "../CounterButton";

it('renders the button', () => {
    expect(shallow(<CounterButton/>)).toMatchSnapshot()
})

it('increments the button', () => {
    const wrapper = shallow(<CounterButton/>)
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.props().color).toEqual(undefined)
})