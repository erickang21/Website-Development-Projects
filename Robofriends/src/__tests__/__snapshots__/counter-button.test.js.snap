// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  0
</button>
`;
